worker/changestream/stream/stream.go

```diff
@@ -206,33 +199,31 @@ func (e changeEvent) ChangedUUID() string {
 
 var errRetryable = errors.New("retryable error")
 
-func (s *Stream) readChanges(stmt *sql.Stmt) ([]changeEvent, error) {
-	rows, err := stmt.Query(s.lastID)
-	if err != nil {
-		if database.IsErrRetryable(err) {
-			s.logger.Debugf("ignoring error during reading changes: %s", err.Error())
-			return nil, errRetryable
-		}
-		return nil, errors.Annotate(err, "querying for changes")
-	}
-	defer rows.Close()
-
+func (s *Stream) readChanges() ([]changeEvent, error) {
 	var changes []changeEvent
-	dest := func(i int) []interface{} {
-		changes = append(changes, changeEvent{})
-		return []interface{}{
-			&changes[i].id,
-			&changes[i].changeType,
-			&changes[i].namespace,
-			&changes[i].changedUUID,
-			&changes[i].createdAt,
+	err := s.db.Txn(context.TODO(), func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
I believe we'll need to change this due to the potential for deadlock discovered yesterday, but it's OK to do in a follow-up.

Reply:
As we're using `s.db.Txn` This will give us a timeout by default, but this isn't currently tied to the tomb. I'll follow your lead when you land your changes, with a follow up PR.